pdata/internal/cmd/pdatagen/internal/base_slices.go

```diff
@@ -240,6 +240,22 @@ func (es ${structName}) Sort(less func(a, b ${elementName}) bool) {
 	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
 }`
 
+const slicePtrTestTemplate = `func Test${structName}_Sort(t *testing.T) {
+	es := generateTest${structName}()
+	es.Sort(func(a, b ${elementName}) bool {
+		return uintptr(unsafe.Pointer(a.orig)) < uintptr(unsafe.Pointer(b.orig))
+	})
+	for i := 1; i < es.Len(); i++ {
+		assert.True(t, uintptr(unsafe.Pointer(es.At(i-1).orig)) < uintptr(unsafe.Pointer(es.At(i).orig)))
```

Comment:
assert.Less?

Reply:
Surprisingly assert.Less doesn't support `uintptr`

Reply:
Trying to fix that :) https://github.com/stretchr/testify/pull/1339

Reply:
Maybe add a TODO to change this if the PR gets merged.

Reply:
Done